src/foam/nanos/session/Session.js

```diff
@@ -233,6 +235,19 @@ foam.CLASS({
           }
           AppConfig appConfig = (AppConfig) x.get("appConfig");
           appConfig = (AppConfig) appConfig.fclone();
+
+          ThemeDomain td = (ThemeDomain) ((DAO) x.get("themeDomainDAO")).find(req.getServerName());
+          if ( td != null ) {
+            Theme theme = (Theme) ((DAO) x.get("themeDAO")).find(td.getTheme());
+            if ( theme != null ) {
+              AppConfig themeAppConfig = theme.getAppConfig();
+              if ( themeAppConfig != null ) {
+                appConfig.copyFrom(themeAppConfig);
+              }
+              rtn = rtn.put("theme", theme);
+            }
+          }
+
```

Comment:
if ThemeDomain or Theme not found, stick a default Theme    'new Theme()' into the context.